### PR TITLE
fix(mongodb): derive ssl usage from connection string

### PR DESCRIPTION
**Pull request by JoergSiebahn**

### CODE REVIEW

File: sda-commons-server-spring-data-mongo/src/main/java/org/sdase/commons/server/spring/data/mongo/MongoConfiguration.java [hunk 1]
```diff
@@ -115,7 +117,7 @@ public MongoConfiguration setHosts(String hosts) {
   @Deprecated(forRemoval = true)
   public String getDatabase() {
     if (StringUtils.isBlank(database) && StringUtils.isNotBlank(connectionString)) {
-      return String.join(",", new ConnectionString(connectionString).getDatabase());
+      return new ConnectionString(connectionString).getDatabase();
```
Comment (christopher-cudennec) [inline sda-commons-server-spring-data-mongo/src/main/java/org/sdase/commons/server/spring/data/mongo/MongoConfiguration.java:120]: If creating the `ConnectionString` is expensive, we could also store it as a field of this class.

File: sda-commons-server-spring-data-mongo/src/test/java/org/sdase/commons/server/spring/data/mongo/MongoConfigurationTest.java [hunk 2]
```diff
@@ -80,4 +80,32 @@ void shouldNotBeValidIfOnlyDatabaseWasSet() {
     MongoConfiguration config = new MongoConfiguration().setDatabase("records");
     assertThat(config.isValid()).isFalse();
   }
+
```
Comment (christopher-cudennec) [inline sda-commons-server-spring-data-mongo/src/test/java/org/sdase/commons/server/spring/data/mongo/MongoConfigurationTest.java:83]: We could also add a test without `ssl` or `tls` in the connection string to ensure that `isUseSsl` is `false`.

File: sda-commons-server-spring-data-mongo/src/test/java/org/sdase/commons/server/spring/data/mongo/SpringDataMongoBundleSSLConfigurationIT.java [hunk 3]
```diff
@@ -44,7 +45,7 @@ class SpringDataMongoBundleSSLConfigurationIT {
           null,
           randomPorts(),
           config("springDataMongo.connectionString", mongo::getConnectionString),
-          config("springDataMongo.useSsl", "false"),
+          config("springDataMongo.useSsl", "true"),
```
Comment (christopher-cudennec) [inline sda-commons-server-spring-data-mongo/src/test/java/org/sdase/commons/server/spring/data/mongo/SpringDataMongoBundleSSLConfigurationIT.java:48]: This value does not match the use case, does it? The app extension's name is `DW_WITHOUT_SSL` but here you set the value to `true`.

I would prefer to not set this deprecated property anymore at all. If the connection string is configured it should override all other settings IMHO.

Comment (JoergSiebahn) [reply sda-commons-server-spring-data-mongo/src/test/java/org/sdase/commons/server/spring/data/mongo/SpringDataMongoBundleSSLConfigurationIT.java:48]: > If the connection string is configured it should override all other settings IMHO.

That's why I changed the test. But I'm not sure if this assumption is correct in all cases as I don't know if the mongo client will upgrade to SSL automatically, if it is not set in the connection string, but the server only offers SSL.

File: sda-commons-server-spring-data-mongo/src/main/java/org/sdase/commons/server/spring/data/mongo/MongoConfiguration.java (hunk 1 above)
Comment (JoergSiebahn) [reply sda-commons-server-spring-data-mongo/src/main/java/org/sdase/commons/server/spring/data/mongo/MongoConfiguration.java:120]: That only happens on startup. Not sure if it's worth it.

File: sda-commons-server-spring-data-mongo/src/test/java/org/sdase/commons/server/spring/data/mongo/MongoConfigurationTest.java (hunk 2 above)
Comment (JoergSiebahn) [reply sda-commons-server-spring-data-mongo/src/test/java/org/sdase/commons/server/spring/data/mongo/MongoConfigurationTest.java:83]: That is tested [here](https://github.com/SDA-SE/sda-dropwizard-commons/pull/2288#discussion_r1166287322), isn't it?.

**Pull request closed without merging**